### ISSUE TITLE
cimpl.KafkaException: KafkaError{code=_INVALID_ARG,val=-186,str="Unsupported value "SASL_SSL" for configuration property "security.protocol": OpenSSL not available at build time"}

### ISSUE THREAD

**Issue by rbhatti999** [confluentinc/confluent-kafka-python#1418]
Description
================

I am on an M1 Mac (Monterey) creating an Avro Python Producer but when I try to produce a message to Confluent Kafka I get this error:

`cimpl.KafkaException: KafkaError{code=_INVALID_ARG,val=-186,str="Unsupported value "SASL_SSL" for configuration property "security.protocol": OpenSSL not available at build time"}`

I tried to install `OpenSSL` with Brew but it was already installed. I then tried to rebuild `librdkafka` with `./configure --reconfigure && make` and I verified that WITH_SSL is set to 1 in `config.h`. I still got the same error. I don't know what else to do. Help is needed ASAP. Thanks.

How to reproduce
================
```
`from configparser import ConfigParser
from confluent_kafka import SerializingProducer
from confluent_kafka.schema_registry import SchemaRegistryClient
from confluent_kafka.schema_registry.avro import AvroSerializer
import json
import uuid

config_parser = ConfigParser()
config_parser.read("config.ini")
topic_name = config_parser["Topic"]["topic.name"]

conf = dict(config_parser["CCloud_Properties"])
schema_registry_conf = {"url": conf["schema.registry.url"],
                        "basic.auth.user.info": conf["basic.auth.user.info"]}

schema_registry_client = SchemaRegistryClient(schema_registry_conf)

email_received_schema = open("EmailReceived.avsc", "r").read()

avro_serializer = AvroSerializer(schema_registry_client, email_received_schema)

conf.pop('schema.registry.url', None)
conf.pop('basic.auth.user.info', None)
conf.pop('basic.auth.credentials.source', None)

producer_conf = conf
producer_conf['value.serializer'] = avro_serializer
producer = SerializingProducer(producer_conf)

record_details = {"count": "4"}

delivered_records = 0

def acked(err, msg):
    global delivered_records
    """Delivery report handler called on
    successful or failed delivery of message
    """
    if err is not None:
        print("Failed to deliver message: {}".format(err))
    else:
        delivered_records += 1
        print("Produced record to topic {} partition [{}] @ offset {}"
              .format(msg.topic(), msg.partition(), msg.offset()))


producer.produce(topic=topic_name, value=record_details, on_delivery=acked)
producer.poll(0)
`
```

Checklist
=========
Please provide the following information:

 - [X] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`): `confluent_kafka.version(1.9.0)` and `confluent_kafka.libversion(1.9.2)`
 - [ ] Apache Kafka broker version:
 - [ ] Client configuration: `{...}`
 - [X] Operating system: MacOS
 - [ ] Provide client logs (with `'debug': '..'` as necessary)
 - [ ] Provide broker log excerpts
 - [X] Critical issue

**Comment by pranavrth** [comment]
Hello @rbhatti999,

Please try to rebuild `librdkafka` with --enable-ssl flag.

`./configure --enable-ssl`

**Comment by rbhatti999** [comment]
Hi @pranavrth,

Thanks for the reply. I ran the command you gave me and I am still getting the exact same error. Is there anything else I can do?

**Comment by pranavrth** [comment]
Have you tried this? https://github.com/confluentinc/confluent-kafka-python#ssl-certificates

**Comment by pranavrth** [comment]
> Hello @rbhatti999,
> 
> Please try to rebuild `librdkafka` with --enable-ssl flag.
> 
> `./configure --enable-ssl`

Can you also make sure you are doing `make` and `make install` after this and building python client again to reflect those changes?

**Comment by rbhatti999** [comment]
I did the following:
- Added this to my code: `conf['ssl.ca.location'] = certifi.where()`
- Ran `make` and that worked. 
- Ran `make install` but got permission denied error.
- Reinstalled `confluent_kafka`

But I still get the same error.

**Comment by pranavrth** [comment]
> I did the following:
> 
> * Added this to my code: `conf['ssl.ca.location'] = certifi.where()`
> * Ran `make` and that worked.
> * Ran `make install` but got permission denied error.
> * Reinstalled `confluent_kafka`
> 
> But I still get the same error.

use `sudo make install`

If the above doesn't work, please try version kafka 1.9.2 wheel. It has everything bundled for M1 mac.

**Comment by rbhatti999** [comment]
I solved my issue. I don't know why this works but I installed Python 3.9 interpreter and then installed confluent_kafka with that interpreter (so basically whenever I use Python 3.9 I don't have to install confluent_kafka again) but when I ran the code using 3.9 it worked. I am guessing I am unintentionally installed the wheels for M1 Mac? idk, but the code worked and I can produce messages now so thanks for all your help!
